src/python/grpcio_reflection/grpc_reflection/v1alpha/proto_reflection_descriptor_database.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2022 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Reference implementation for reflection client in gRPC Python."""
+
+from google.protobuf.descriptor_database import DescriptorDatabase
+from google.protobuf.descriptor_pb2 import FileDescriptorProto
+import grpc
+from grpc_reflection.v1alpha.reflection_pb2 import (
+    ErrorResponse,
+    ExtensionNumberResponse,
+    ExtensionRequest,
+    FileDescriptorResponse,
+    ListServiceResponse,
+    ServerReflectionRequest,
+    ServerReflectionResponse,
+    ServiceResponse,
+)
+from grpc_reflection.v1alpha.reflection_pb2_grpc import ServerReflectionStub
+import logging
+from typing import Iterable, Any
+
+
+class ProtoReflectionDescriptorDatabase(DescriptorDatabase):
+    """
+    ProtoReflectionDescriptorDatabase takes a stub of ServerReflection and provides the methods defined by
+    DescriptorDatabase interfaces. It can be used to feed a DescriptorPool instance.
+
+    Python implementation based on C++ version found here:
+      https://github.com/grpc/grpc/blob/v1.39.1/test/cpp/util/proto_reflection_descriptor_database.cc
+    while implementing the interface given here:
+      https://googleapis.dev/python/protobuf/3.17.0/google/protobuf/descriptor_database.html
+
+    """
+
+    def __init__(self, channel: grpc.Channel):
+        DescriptorDatabase.__init__(self)
+        self._logger = logging.getLogger(__name__)
+        self._stub = ServerReflectionStub(channel)
+        self._known_files: Set[str] = set()
+        self._cached_extension_numbers: Dict[str, List[int]] = dict()
+
+    def get_services(self) -> Iterable[str]:
```

Comment:
For all public API, can you add docstring for all of them, following the Google style guide, with arguments, returns.

Reply:
Done